src/App.vue

```diff
@@ -107,6 +109,30 @@ export default {
     },
   },
   watch: {
+    async $route() {
+      // A routing event has just occurred, which is either the initial page load or a subsequent
+      // navigation. So load any custom scripts that should be run, based on their `run` property,
+      // after this routing event.
+      //
+      // This hook, and (as a result) any appropriate custom scripts for the current routing event,
+      // are called *after* the HTML for the current route has been dynamically added to the DOM.
+      // This means that custom scripts have access to the documentation HTML for the current
+      // topic (or tutorial, etc).
+
+      if (this.initialRoutingEventHasOccurred) {
+        // The initial page load counts as a routing event, so we only want to run "on-navigate"
+        // scripts from the second routing event onward.
+        await runCustomNavigateScripts();
+      } else {
+        // The "on-load" scripts are run here (on the routing hook), not on `created` or `mounted`,
+        // so that the scripts have access to the dynamically-added documentation HTML for the
+        // current topic.
+        await runCustomPageLoadScripts();
```

Comment:
The visual issue you’re describing in scenario 2 is a quirk of alerts: despite what the browser may be displaying, the alert in scenario 2 _is_ fired strictly after the link is clicked and the subpage is loaded. To prove this, try the following script instead:
```js
window.alert(document.querySelector('h1').innerText);
```
and notice that the alert text matches the H1 of the subpage to which you’re navigating, not of the subpage you’re in, regardless of what the browser is displaying. This means that, in scenario 2, the script _is_ running strictly after navigation.

As an additional test, also try:
```js
document.querySelector('h1').style.fontStyle = 'italic';
```
and notice, in scenario 2, that the H1 (dynamically inserted into the subpage to which you're navigating) is successfully styled, which means that the script did run strictly after navigation.

As for using vue-router hooks instead of the `$route` watcher, I agree that that would be the more correct approach. I’ll work on it.

Reply:
> The visual issue you’re describing in scenario 2 is a quirk of alerts: despite what the browser may be displaying, the alert in scenario 2 _is_ fired strictly after the link is clicked and the subpage is loaded. To prove this, try the following script instead:
> 
> ```js
> window.alert(document.querySelector('h1').innerText);
> ```
> 
> and notice that the alert text matches the H1 of the subpage to which you’re navigating, not of the subpage you’re in, regardless of what the browser is displaying. This means that, in scenario 2, the script _is_ running strictly after navigation.
> 
> As an additional test, also try:
> 
> ```js
> document.querySelector('h1').style.fontStyle = 'italic';
> ```
> 
> and notice, in scenario 2, that the H1 (dynamically inserted into the subpage to which you're navigating) is successfully styled, which means that the script did run strictly after navigation.

Whoops. Got it, that makes sense.

> As for using vue-router hooks instead of the `$route` watcher, I agree that that would be the more correct approach. I’ll work on it.

Sounds good. I think this approach is fine after realizing my problem was with my example, although it still might be good to use those hooks if possible to be safe from any potential issues related to the navigation cycle.